avocado/utils/vmimage.py

```diff
@@ -351,10 +351,10 @@ def list_providers():
     """
     List the available Image Providers
     """
-    return [_ for _ in globals().itervalues()
-            if (_ != ImageProviderBase and
-                isinstance(_, type) and
-                issubclass(_, ImageProviderBase))]
+    return set(_ for _ in globals().itervalues()
+               if (_ != ImageProviderBase and
+                   isinstance(_, type) and
+                   issubclass(_, ImageProviderBase)))
```

Comment:
I lack context about the element instances. Is every instance guaranteed to be different, or the instances have an `__eq__` method implemented? Because if we can instantiate multiple image providers with the exact same parameters, the use of `set` wouldn't work, given that every instance is a different object anyway.

Reply:
We don't register instances, but classes (`isinstance(_, type)`) so the classes should match:

```python
>>> from avocado.utils import vmimage
>>> len(vmimage.IMAGE_PROVIDERS)
5
>>> vmimage.IMAGE_PROVIDERS.add(vmimage.FedoraImageProvider)
>>> len(vmimage.IMAGE_PROVIDERS)
5
```

Reply:
Ah, I thought it was the instance, not the class itself that was being put there. My mistake, looks good!